app/scripts/background.js

```diff
@@ -191,7 +192,13 @@ browser.runtime.onConnectExternal.addListener(async (...args) => {
   // Queue up connection attempts here, waiting until after initialization
   await isInitialized;
   // This is set in `setupController`, which is called as part of initialization
-  connectExternal(...args);
+  const port = args[0];
+
+  if (port.sender.tab?.id && process.env.BARAD_DUR) {
```

Comment:
> port.sender.tab?.id

This is in this block to filter connections from other extensions?

Reply:
it's to treat externally_connectable connections made from extensions and webpages differently